src/lsm/lsm_cursor.c

```diff
@@ -264,6 +264,13 @@ __clsm_leave(WT_CURSOR_LSM *clsm)
     if (F_ISSET(clsm, WT_CLSM_ACTIVE)) {
         --session->ncursors;
         __cursor_leave(session);
+        /*
```

Comment:
WT-6339 misses this usage of __cursor_leave

Reply:
Part of the reason the original change was preferred was that it meant all usages of __cursor_leave were consistent, and that we didn't need to think about LSM cursors being different and having an odd piece of code about releasing transaction snapshots buried inside the LSM cursor implementation.